bin/ch/WScriptJsrt.cpp

```diff
@@ -225,33 +238,39 @@ JsValueRef WScriptJsrt::LoadScriptHelper(JsValueRef callee, bool isConstructCall
     return returnValue;
 }
 
-JsValueRef WScriptJsrt::LoadScript(JsValueRef callee, LPCWSTR fileName, size_t fileNameLength, LPCWSTR fileContent, LPCWSTR scriptInjectType, bool isSourceModule)
+JsValueRef WScriptJsrt::LoadScript(JsValueRef callee, LPCSTR fileName, LPCWSTR fileContent, LPCWSTR scriptInjectType, bool isSourceModule)
 {
     HRESULT hr = E_FAIL;
     JsErrorCode errorCode = JsNoError;
-    LPCWSTR errorMessage = L"Internal error.";
+    LPCWSTR errorMessage = _u("Internal error.");
     size_t errorMessageLength = wcslen(errorMessage);
     JsValueRef returnValue = JS_INVALID_REFERENCE;
     JsErrorCode innerErrorCode = JsNoError;
     JsContextRef currentContext = JS_INVALID_REFERENCE;
     JsRuntimeHandle runtime = JS_INVALID_RUNTIME_HANDLE;
-
+    
+    wchar_t* fullPath = nullptr;
+    char fullPathNarrow[_MAX_PATH];
+    size_t len = 0;
+    
     IfJsrtErrorSetGo(ChakraRTInterface::JsGetCurrentContext(&currentContext));
     IfJsrtErrorSetGo(ChakraRTInterface::JsGetRuntime(currentContext, &runtime));
 
-    wchar_t fullPath[_MAX_PATH];
-    if (_wfullpath(fullPath, fileName, _MAX_PATH) == nullptr)
+    if (_fullpath(fullPathNarrow, fileName, _MAX_PATH) == nullptr)
     {
         IfFailGo(E_FAIL);
     }
     // canonicalize that path name to lower case for the profile storage
-    size_t len = wcslen(fullPath);
+    // REVIEW: This doesn't work for UTF8...
+    len = strlen(fullPathNarrow);
```

Comment:
Actually, the comment was less about using strlen and more about how we use the length to convert the string to its lower-case equivalent. This approach doesn't work with UTF8 strings- since it's null-terminated, we could do a single pass conversion (read a char, convert to lower case, replace) until we hit the null char. However, not sure if we can do it in-place in all cases or if there are cases where the lower case version of a character takes up more bytes than the upper case one. Anyway, it's academic for now which is why I left it as a review comment in case anyone actually runs into the issue using an UTF8 path to a file :)

Side note: Not sure if _mbstrlen works with UTF8 strings (I think it predates utf8 and the mb prefix refers to some other multi-byte encodings but could be wrong)